### PR TITLE
Eth sign typed data

### PR DESCRIPTION
**Pull request by estebanmino**
This PR is mostly a migration to ts of [this](https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/lib/typed-message-manager.js) to have both `eth_signTypedData` and `eth_signTypedData_v3` as proposed by EIP712

It also exposes signing methods from eth-sig-util

### CODE REVIEW

File: src/KeyringController.ts [hunk 1]
```diff
@@ -258,6 +258,29 @@ export class KeyringController extends BaseController<BaseConfig, KeyringState>
...
+	 */
+	async signTypedMessage(messageParams: MessageParams, version: string) {
+		try {
+			const address = sigUtil.normalize(messageParams.from);
+			const privateKey = await this.exportAccount(address);
+			const privateKeyBuffer = ethUtil.toBuffer(ethUtil.addHexPrefix(privateKey));
+			switch (version) {
+				case 'V1':
+					return sigUtil.signTypedDataLegacy(privateKeyBuffer, { data: messageParams.data });
+				case 'V3':
```
Comment (brunobar79) [inline src/KeyringController.ts]: should we make version optional and default to v3?

File: src/TypedMessageManager.test.ts [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+		name: 'Message',
+		type: 'string',
+		value: 'Hi, Alice!'
+	},
+	{
+		name: 'A number',
+		type: 'uint32',
+		value: '1337'
+	}
+];
```
Comment (brunobar79) [inline src/TypedMessageManager.test.ts]: Is it worth to add a test for a message with data that has >= 2 levels deep ?

File: src/KeyringController.ts (hunk 1 above)
Comment (estebanmino) [reply src/KeyringController.ts]: good idea, what do you think @bitpshr ? This was the way it is [here](https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/lib/typed-message-manager.js)

File: src/TypedMessageManager.test.ts (hunk 2 above)
Comment (brunobar79) [reply src/TypedMessageManager.test.ts]: Just making sure that works for complex objects

Comment (estebanmino) [reply src/TypedMessageManager.test.ts]: The way I understand it, these managers only have the responsibility to handle messages flow and not the data itself, that would be a responsibility for `eth-sig-util` in this case. But I can be wrong and we could add more complex data structures extending from `object[]`. For that reason there are also tests for `eth_signTypedData_v3` with typed messages validation according to the EIP712.

**Pull request merged**